livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -161,6 +162,54 @@ def with_cerebras(
             temperature=temperature,
         )
 
+    @staticmethod
+    def with_vertexai(
+        *,
+        model: str | GeminiModels = "gemini-1.5-flash-002",
+        project_id: str | None = None,
+        location: str | None = "us-central1",
+        client: openai.AsyncClient | None = None,
+        user: str | None = None,
+        temperature: float | None = None,
+    ) -> LLM:
+        """
+        Create a new instance of VertexAI LLM.
+
+        ``project_id`` must be set to your VERTEXAI PROJECT ID, either using the argument or by setting
+        the ``VERTEXAI_PROJECT_ID`` environmental variable.
+        """
+
+        project_id = project_id or os.environ.get("VERTEXAI_PROJECT_ID")
+        if project_id is None:
+            raise ValueError(
+                "VERTEXAI_PROJECT_ID is required, either set project_id argument or set VERTEXAI_PROJECT_ID environmental variable"
+            )
+        location = location or os.environ.get("VERTEXAI_LOCATION")
+        if location is None:
+            raise ValueError(
+                "VERTEXAI_LOCATION is required, either set location argument or set VERTEXAI_LOCATION environmental variable"
+            )
+
+        from google.auth import default
+        from google.auth.transport import requests
+
+        credentials, _ = default(
+            scopes=["https://www.googleapis.com/auth/cloud-platform"]
+        )
+        auth_request = requests.Request()
+        credentials.refresh(auth_request)
+        base_url = f"https://{location}-aiplatform.googleapis.com/v1beta1/projects/{project_id}/locations/{location}/endpoints/openapi"
+        api_key = credentials.token
```

Comment:
What is the TTL of this token?

Reply:
We should automatically refresh it, otherwise the LLM instance could start to fail after this TTL